include/tscpp/api/Plugin.h

```diff
@@ -142,6 +143,15 @@ class Plugin : noncopyable
     transaction.resume();
   };
 
+  /**
+   * This method must be implemented when you hook HOOK_TXN_CLOSE
+   */
+  virtual void
+  handleTxnClose(Transaction &transaction)
+  {
+    transaction.resume();
```

Comment:
You would not want to call resume here (leave this function empty) if you take my suggestion to move the call to this member function to cleanupTransactionPlugin() .